### PR TITLE
Feature/node maintenance mode

### PR DESCRIPTION
**Pull request by iStrike7**

### CODE REVIEW

File: pallets/dmp-queue/src/lib.rs [hunk 1]
```diff
@@ -123,6 +126,8 @@ pub mod pallet {
 		Unknown,
 		/// The amount of weight given is possibly not enough for executing the message.
 		OverLimit,
+		/// Dmp message processing is blocked by maintenance mode
+		DmpMsgProcessingBlockedByMaintenanceMode
```
Comment (mateuszaaa) [inline pallets/dmp-queue/src/lib.rs]: what Dmp means ?

File: pallets/xcmp-queue/src/lib.rs [hunk 2]
```diff
@@ -141,6 +144,8 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			T::ExecuteOverweightOrigin::ensure_origin(origin)?;
 
+			ensure!(!T::MaintenanceStatusProvider::is_maintenance(), Error::<T>::XcmMsgProcessingBlockedByMaintenanceMode);
```
Comment (mateuszaaa) [inline pallets/xcmp-queue/src/lib.rs]: should that rather early return instead ?

**Review by mateuszaaa — COMMENTED** [review]
just few questions

File: pallets/dmp-queue/src/lib.rs (hunk 1 above)
Comment (iStrike7) [reply pallets/dmp-queue/src/lib.rs]: DownwardMessageQueue

File: pallets/xcmp-queue/src/lib.rs (hunk 2 above)
Comment (iStrike7) [reply pallets/xcmp-queue/src/lib.rs]: It is returning early, right? Or do you mean even before the ensure_origin check?

**Pull request merged**